common/scala/src/main/scala/whisk/core/entity/WhiskStore.scala

```diff
@@ -69,6 +69,14 @@ protected[core] trait WhiskDocument extends DocumentSerializer with DocumentRevi
    */
   protected[core] final def docinfo: DocInfo = DocInfo(docid, rev)
 
+  /**
+   * Determines entity type based on entity class name
+   */
+  protected[core] def documentType: String = {
+    val typeName = this.getClass.getSimpleName.toLowerCase
+    if (typeName.startsWith("whisk")) typeName.substring(5) else typeName
+  }
```

Comment:
Should we rather override this property in the specific subtypes instead of relying on "magic"?

Alternatively, maybe we can get away with something like:

```scala
protected[core] def documentType: String = this match {
  case _: WhiskAction => "action"
  case _: WhiskRule => "rule"
  ...
}
```

Didn't test this though.

Reply:
Ack. Would work on a new PR